components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/APIDefinitionFromSwagger12.java

```diff
@@ -0,0 +1,418 @@
+
+        JSONParser parser = new JSONParser();
+        JSONObject apiJSON = null;
+        try {
+            if (registry.resourceExists(resourcePath + APIConstants.API_DOC_1_2_RESOURCE_NAME)) {
+                Resource apiDocResource = registry.get(resourcePath + APIConstants.API_DOC_1_2_RESOURCE_NAME);
+                String apiDocContent = new String((byte[]) apiDocResource.getContent());
+                apiJSON = (JSONObject) parser.parse(apiDocContent);
+                JSONArray pathConfigs = (JSONArray) apiJSON.get("apis");
+
+                for (int k = 0; k < pathConfigs.size(); k++) {
+                    JSONObject pathConfig = (JSONObject) pathConfigs.get(k);
+                    String pathName = (String) pathConfig.get("path");
+                    pathName = pathName.startsWith("/") ? pathName : ("/" + pathName);
+
+                    Resource pathResource = registry.get(resourcePath + pathName);
+                    String pathContent = new String((byte[]) pathResource.getContent());
+                    JSONObject pathJSON = (JSONObject) parser.parse(pathContent);
+                    pathConfig.put("file", pathJSON);
+                }
+            }
+        } catch (RegistryException e) {
+            handleException("Error while retrieving Swagger Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion(), e);
+        } catch (ParseException e) {
+            handleException("Error while parsing Swagger Definition for " + apiIdentifier.getApiName() + "-" +
+                    apiIdentifier.getVersion() + " in " + resourcePath, e);
+        }
+        return apiJSON.toJSONString();
+
+    }
+
+    @Override
+    @Deprecated
+    public String createAPIDefinition(API api) throws APIManagementException {
+        JSONParser parser = new JSONParser();
+        String pathJsonTemplate = "{\n    \"path\": \"\",\n    \"operations\": []\n}";
+        String operationJsonTemplate = "{\n    \"method\": \"\",\n    \"parameters\": []\n}";
+        String apiJsonTemplate = "{\n    \"apiVersion\": \"\",\n    \"swaggerVersion\": \"1.2\",\n    \"apis\": [],\n    \"info\": {\n        \"title\": \"\",\n        \"description\": \"\",\n        \"termsOfServiceUrl\": \"\",\n        \"contact\": \"\",\n        \"license\": \"\",\n        \"licenseUrl\": \"\"\n    },\n    \"authorizations\": {\n        \"oauth2\": {\n            \"type\": \"oauth2\",\n            \"scopes\": []\n        }\n    }\n}";
+        String apiResourceJsontemplate = "{\n    \"apiVersion\": \"\",\n    \"swaggerVersion\": \"1.2\",\n    \"resourcePath\":\"\",\n    \"apis\": [],\n    \"info\": {\n        \"title\": \"\",\n        \"description\": \"\",\n        \"termsOfServiceUrl\": \"\",\n        \"contact\": \"\",\n        \"license\": \"\",\n        \"licenseUrl\": \"\"\n    },\n    \"authorizations\": {\n        \"oauth2\": {\n            \"type\": \"oauth2\",\n            \"scopes\": []\n        }\n    }\n}";
+
```

Comment:
APIDefinitionFromSwagger12 is used with swagger v1.2(In APIM 1.8.0) and removed since it is no longer in use(In APIM 1.9.0)
Instead we have APIDefinitionFromSwagger20 Implemented from same abstract class which implements APIDefinitionFromSwagger12